### PR TITLE
Validate that all required gcc switches were used

### PR DESCRIPTION
**Pull request by ppannuto**
This is conservative right now and only opportunistically checks if
it can find the elf file, which I think is the right behavoir, as it
supports concepts like passing binaries around for others to flash
without getting in the way, but will likely protect users in the common
case.

We could add a warning print or s/t if the .elf isn't found, but I'm
not sure that passes muster for SNR.

### CODE REVIEW

File: tockloader/main.py [hunk 1]
```diff
@@ -801,6 +802,48 @@ def collect_binaries (binaries, single=False):
 			print('No binaries found.')
 			sys.exit(1)
 
+		# Opportunistically match the .elf files and validate they were built
+		# with all the flags that Tock applications require
+		tock_flags = ('-msingle-pic-base', '-mpic-register=r9', '-mno-pic-data-is-text-relative')
+		if not args.no_check_switches:
+			for binfile in binaries:
+				print(binfile)
```
Comment (bradjc) [inline tockloader/main.py]: remove

File: tockloader/main.py [hunk 2]
```diff
@@ -801,6 +802,48 @@ def collect_binaries (binaries, single=False):
...
+				if binfile[-4:] == '.bin':
+					elffile = binfile[:-4] + '.elf'
+					print(elffile)
```
Comment (bradjc) [inline tockloader/main.py]: remove

File: tockloader/main.py [hunk 3]
```diff
@@ -801,6 +802,48 @@ def collect_binaries (binaries, single=False):
...
+					if os.path.exists(elffile):
+						p = subprocess.Popen(['arm-none-eabi-readelf',
+								'-p', '.GCC.command.line', elffile],
+								stdout=subprocess.PIPE,
+								stderr=subprocess.PIPE)
+						out, err = p.communicate()
+						if 'does not exist' in err.decode('utf-8'):
+							print("Error: Missing section .GCC.command.line in " + elffile)
```
Comment (bradjc) [inline tockloader/main.py]: Use `'` instead of `"`

**Comment by ppannuto** [comment]
fixed

**Pull request merged**